src/execution.rs

```diff
@@ -320,16 +411,51 @@ impl Stanza {
                 shorthands: shorthands,
             };
             for statement in &self.statements {
-                statement
-                    .execute(&mut exec)
-                    .with_context(|| format!("Executing {}", statement))?;
+                statement.execute(&mut exec).or_else(|e| {
+                    Err(e).with_context(|| {
+                        let node = mat
+                            .captures
+                            .iter()
+                            .find(|c| c.index as usize == self.full_match_file_capture_index)
+                            .unwrap()
+                            .node;
+                        format!(
+                            "While executing statement {}\n{}\nin stanza\n{}\nmatching ({}) node\n{}",
+                            statement,
+                            Excerpt::from_source(tsg_path, tsg_source, statement.location()),
```

Comment:
We're using `Location`s instead of `Range`s (that is, these were already in use prior to this PR), so we only print the one line. Future work if we want `clang`-style diagnostics, for example.